File: Week1/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    int c = 0;
    float cash;
    // ask for cash owed
    do
    {
        cash = get_float("Change owed: ");
    }
    while (cash < 0);
    // convert cash to change
    int change = round(cash * 100);
    // take out quarters
    while (change  >= 25)
    {
        change = change - 25;
        c++;
    }
    // take out dimes
    while (change >= 10)
    {
        change = change - 10;
        c++;
    }
    // take out nickles
    while (change >= 5)
    {
        change = change - 5;
        c++;
    }
    // take out pennies
    c = c + change;
    // prints coins
    printf("%i\n", c);
}

File: Week6/pset6/cash/cash.py
from cs50 import get_float


# Prompt user for change
while True:
    change = get_float("Change owed: ")
    if change > 0:
        break
    
change = change * 100  # Convert change to int to avoid float imprecision
counter = 0  # Counts number of coins 

# Checks quarters
while change >= 25:
    counter += 1
    change -= 25

# Checks dimes
while change >= 10:
    counter += 1
    change -= 10

# Checks nickles
while change >= 5:
    counter += 1
    change -= 5

# Checks pennies
while change >= 1:
    counter += 1
    change -= 1

# Prints coins used
print(counter)